src/common/Makefile.am

```diff
@@ -100,7 +100,10 @@ noinst_LTLIBRARIES += libcommon_internal.la
 libcommon_api_la_SOURCES = \
 	common/buffer.cc
 if LINUX
+if !ENABLE_XIO
+# XioMessenger.cc uses get_xio_mp() in buffer.cc
 libcommon_api_la_CXXFLAGS = -fvisibility=hidden -fvisibility-inlines-hidden
+endif # !ENABLE_XIO
```

Comment:
thanks for your insights @gregsfortytwo , let me think about again to see if we have any other way to do it.